### DISCUSSION TITLE
Is it possible to exclude instances from job processing?

### DISCUSSION THREAD

**Discussion by mateusz353** [kagkarlsson/db-scheduler#463] [Q&A]
Hi, sometimes I need to run the service locally with test environment database, and then my application starts processing jobs. Sometimes local version is  incompatible with the instances from test environment. It case problem with testing. Is it possible to exclude instances from job processing and still be able to scheduling new tasks? Some property?

If not, I would suggest 2 solutions:
```
db-scheduler:
  enabled: true
  polling-enabled: false
```
or
```
db-scheduler:
  enabled: true
  polling-strategy: NONE
```

**Comment by kagkarlsson** [comment]
You can instantiate only a `SchedulerClient`

**Reply by roshan092** [reply]
My use case was multiple devs connecting to the dev environment DB instead of their local DB. The scheduler used to run on their local and the tasks failed, causing unexpected issues on dev. Stumbled on this discussion which helped me come up with a solution. If it helps anyone in future, the solution we used for sprint boot auto configuration is:
Set `db-scheduler.enabled: false` on local
Add configuration for missing beans:
```
@Configuration
@ConditionalOnProperty(value = "db-scheduler.enabled", havingValue = "false")
public class DBSchedulerLocalConfig {

  @Bean
  public SchedulerClient schedulerClient(DataSource dataSource) {
    return Scheduler.create(dataSource).threads(5).build();
  }

  @Bean
  StatsRegistry noopStatsRegistry() {
    return StatsRegistry.NOOP;
  }

  @Bean
  public DbSchedulerCustomizer noopCustomizer() {
    return new DbSchedulerCustomizer() {};
  }
}
```

**Reply by kagkarlsson** [reply]
Nice workaround 👍
